lib/utility/errors.rb

```diff
@@ -119,8 +119,8 @@ def to_h
   end
 
   class HealthCheckFailedError < StandardError
-    def initialize
-      super('Health check failed for 3rd-party service.')
+    def initialize(msg = nil)
+      super("Health check failed for 3rd-party service: #{msg}")
```

Comment:
If `direct_connection` is not a valid bool value, it will fail here first
https://github.com/elastic/connectors-ruby/blob/aac87fd740f52e965480af5936167f33589640ad/lib/core/sync_job_runner.rb#L54

before it reaches 
https://github.com/elastic/connectors-ruby/blob/aac87fd740f52e965480af5936167f33589640ad/lib/core/sync_job_runner.rb#L61

I think this change also makes sense even without this use case. A hardcoded error message `Health check failed for 3rd-party service` is not very helpful and actionable.